R/align_dia_runs.R

```diff
@@ -410,6 +429,14 @@ perBatch <- function(iBatch, peptides, multipeptide, refRuns, precursors, prec2c
     ref <- refRuns[rownum, "run"][[1]]
     idx <- (rownum - (iBatch-1)*batchSize)
     analytes <- analytesA[[idx]]
+    if ( is.null(multiFeatureAlignmentMap) )
+    {
+      feature_alignment_map <- NULL
+    }
+    else
+    {
+      feature_alignment_map <- multiFeatureAlignmentMap[[rownum]]
```

Comment:
rownum indicates a peptide not a precursor, when you match peaks (based on time) then two peaks will be matched from multiple charge-states, you should be keeping both but since I guess you are using minimum that would pick only one?

Reply:
Yes, but I noted that `alignToRef` currently uses the highest quality precursor, [align_dia_runs.R#L537-L546](https://github.com/singjc/DIAlignR/blob/6ec404cef32d65c1febedcc2e0de25522ccbb479/R/align_dia_runs.R#L537-L546) and [align_dia_runs.R#L581-L582](https://github.com/singjc/DIAlignR/blob/6ec404cef32d65c1febedcc2e0de25522ccbb479/R/align_dia_runs.R#L581-L582).